core/include/irods/private/http_api/common.hpp

```diff
@@ -143,6 +144,8 @@ namespace irods::http
 	auto parse_url(const request_type& _req) -> url;
 
 	auto resolve_client_identity(const request_type& _req) -> client_identity_resolution_result;
+
+	auto get_port_from_url(boost::urls::url_view _url) -> std::optional<std::string>;
```

Comment:
Is it recommended to copy `url_view`?

Reply:
Since it's a view, it holds a non-owning ref/pointer to the data. This should make it inexpensive to copy, like `string_view`.

A skim of the docs doesn't show a preference for either one.

Reply:
Works for me.